include/boost/math/differentiation/autodiff.hpp

```diff
@@ -727,10 +741,10 @@ fvar<RealType,Order>::operator+(const fvar<RealType2,Order2>& cr) const
   for (size_t i=0 ; i<=(std::min)(Order,Order2) ; ++i)
     retval.v[i] = v[i] + cr.v[i];
   if BOOST_AUTODIFF_IF_CONSTEXPR (Order < Order2)
-    for (size_t i=Order+1 ; i<=Order2 ; ++i)
+    for (auto i=Order+1 ; i<=Order2 ; ++i)
       retval.v[i] = cr.v[i];
   else if BOOST_AUTODIFF_IF_CONSTEXPR (Order2 < Order)
-    for (size_t i=Order2+1 ; i<=Order ; ++i)
+    for (auto i=Order2+1 ; i<=Order ; ++i)
```

Comment:
Clang-Tidy/ReSharper, since in all cases where an `auto` was recommended, the type was obvious (ie `Order2` is a `std::size_t`), but I believe the origin would be Herber Sutter's AAA. I believe I avoided it in cases like `auto i = 0;`, since in such cases, we would want to specify the type of `i`.

I have no personal attachment to `auto` (except in cases where the variability of the type makes code more simple to write and maintain), but I used `auto` in these cases since they would likely become noise when using code quality tools.

Reply:
No need to revert, but fyi I'm not a fan of AAA. IMO it's ok when implementation is cleanly separated from the API, but in many cases it isn't (e.g. we can have to cast between types) and in these cases AAA gets in the way.

Reply:
`autos` removed for clarity